ssl/statem/extensions.c

```diff
@@ -1078,7 +1078,7 @@ static int final_key_share(SSL *s, unsigned int context, int sent, int *al)
             && (!s->hit
                 || (s->ext.psk_kex_mode & TLSEXT_KEX_MODE_FLAG_KE) == 0)) {
         /* Nothing left we can do - just fail */
-        *al = SSL_AD_HANDSHAKE_FAILURE;
+        *al = SSL_AD_MISSING_EXTENSION;
```

Comment:
No, the "final" functions are called at the end of the parsing stage. So, "sent" here is true if the extension was sent to us by the peer.